plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/M2RepositoryService.java

```diff
@@ -109,4 +109,16 @@ protected void activate() {
   void setMavenHelper(MavenHelper mavenHelper) {
     this.mavenHelper = mavenHelper;
   }
+
+  /**
+   * First checks if the artifact is available locally. If not, then it tries to resolve it via
+   * Maven.
+   */
+  @Override
+  public void checkArtifactAvailable(LibraryFile libraryFile, IProgressMonitor monitor) throws CoreException {
```

Comment:
+1, since this creates side-effect. Perhaps this also applies to `checkAppEngineStandard` and `checkRuntimeAvailability`. The Javadoc of `checkRuntimeAvailability` could be fixed too to clarify that it prepares necessary artifacts.